circuit/environment/Cargo.toml

```diff
@@ -59,6 +59,9 @@ version = "0.2"
 [dependencies.once_cell]
 version = "1.18.0"
 
+[dependencies.singlevec]
```

Comment:
👁️ 👁️ 
<img width="319" alt="Screenshot 2025-03-08 at 21 34 11" src="https://github.com/user-attachments/assets/66ca46b9-88b3-4e0d-b5a5-f2b5f644e592" />

How big would the diff be if we just use tinyvec? Do snark{VM, OS} have other dependencies which are only maintained by 1-2 folks? I only know of `deadline`. In such cases, I'm thinking vendoring might be the better way forward so we take full ownership over the auditing...

Reply:
I'm happy to try out `tinyvec` as well, but I suspect it would also have worse performance (as long as it doesn't special-case a single-element collection).
The `singlevec` crate is very simple (in fact, a lot simpler than `smallvec` or `tinyvec`), IIRC it's basically a single file; the reason why I'm listed as a contributor is that I needed `insert` and `remove` methods that weren't there at the time, and it took me only a few minutes to include them.